contracts/token/ERC721/extensions/draft-ERC721Royalty.sol

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts v4.4.0 (token/ERC721/extensions/draft-ERC721Royalty.sol)
+
+pragma solidity ^0.8.0;
+
+import "../ERC721.sol";
+import "./draft-IERC721Royalty.sol";
+import "../../../utils/introspection/ERC165.sol";
+
+/**
+ * @dev Implementation of the ERC721 Royalty extension allowing royalty information to be stored and retrieved, as defined in
+ * https://eips.ethereum.org/EIPS/eip-2981[EIP-2981].
+ *
+ * Adds the {_setTokenRoyalty} methods to set the token royalty information, and {_setGlobalRoyalty} method to set a global
+ * royalty information.
+ *
+ * _Available since v4.5._
+ */
+abstract contract ERC721Royalty is IERC721Royalty, ERC721 {
+    struct RoyaltyInfo {
+        address receiver;
+        uint256 royaltyFraction;
```

Comment:
if the denominator is 10000, then we could possibly use a uint96 so that the info fit into a single slot